gh-77214: Update outdated documentation for numeric PyArg_Parse formats (#128454)
python/cpython: Doc/c-api/arg.rst

Numbers
-------

> These formats allow to represent Python numbers or single characters as C numbers.

Reviewer: ```suggestion
These formats allow representing Python numbers or single characters as C numbers.
```

Formats that require :class:`int`, :class:`float` or :class:`complex` can
also use the corresponding special methods :meth:`~object.__index__`,
:meth:`~object.__float__` or :meth:`~object.__complex__` to convert
the Python object to the required type.

> For signed integer format, :exc:`OverflowError` is raised if the value

Reviewer 2: ```suggestion
For signed integer formats, :exc:`OverflowError` is raised if the value
```

is out of range for the C type.

> For unsigned integer format, no proper range checking is done --- the

Reviewer 2: ```suggestion
For unsigned integer formats, no proper range checking is done --- the
```

Reviewer 2: Also should it not just be `--` and not `---` for em dash?
```suggestion
For unsigned integer formats, no proper range checking is done -- the
```

> For unsigned integer formats, no proper range checking is done --- the

Reviewer: ```suggestion
For unsigned integer formats, no range checking is done --- the
```
What's "proper" depends on the use case.
(IMO, in the deleted paragraph below, both “It is possible to pass "long" integers” and “proper” are an artifact of the long-gone distinction between `int` & `long` Python types.)
